### PR TITLE
Doc: Added a proposal for swift-etcd-client API

### PR DESCRIPTION
**Pull request by ayushi2103**
A detailed design for a native Swift-etcd client's get function that aims to seamlessly integrate etcd, a distributed key-value store into Swift applications.

### CODE REVIEW

File: Sources/ETCD/Docs.docc/Proposals/0001-swift-etcd-api.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+- Get: Unary RPC 
+    - retrieves the value associated with a given key from the etcd server
+
+### Key-Value Operations
+
+- An abstract base class providing a base implementation for key-value operations and key range queries and bulk deletions.
+
+```swift
+/// Abstract class providing a base implementation for key-value get operation with etcd
+class BaseEtcdClient {
```
Comment (FranzBusch) [inline Sources/ETCD/Docs.docc/Proposals/0001-swift-etcd-api.md]:
```suggestion
fina class ETCDClient {
```

File: Sources/ETCD/Docs.docc/Proposals/0001-swift-etcd-api.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+## Detailed Design
+
+The API provides methods for interacting with etcd's key-value store, covering the following operation:
+
+- Get: Unary RPC 
+    - retrieves the value associated with a given key from the etcd server
+
+### Key-Value Operations
+
+- An abstract base class providing a base implementation for key-value operations and key range queries and bulk deletions.
```
Comment (FranzBusch) [inline Sources/ETCD/Docs.docc/Proposals/0001-swift-etcd-api.md]: This doesn't have to be an abstract class

File: Sources/ETCD/Docs.docc/Proposals/0001-swift-etcd-api.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+```swift
+/// Abstract class providing a base implementation for key-value get operation with etcd
+class BaseEtcdClient {
+    /// Method to get value for a key from etcd
+    public func get<Value: etcdClientValue>(_ key: String, as valueType: Value.Type = Value.self) async throws -> Value?  {
```
Comment (FranzBusch) [inline Sources/ETCD/Docs.docc/Proposals/0001-swift-etcd-api.md]: Can we please include the definition of the `ECTDValue` protocol in this API spec
```suggestion
    public func get<Value: ETCDValue>(_ key: String, as valueType: Value.Type = Value.self) async throws -> Value?  {
```

File: Sources/ETCD/Docs.docc/Proposals/0001-swift-etcd-api.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        /// Implementation to retrieve value from etcd
+        return nil
```
Comment (FranzBusch) [inline Sources/ETCD/Docs.docc/Proposals/0001-swift-etcd-api.md]: We don't have to mention the implementation in the API spec here

**Comment by ayushi2103** [comment]
Closing the PR as not required at the moment.

**Pull request closed without merging**